vhdbuilder/packer/test/run-test.sh

```diff
@@ -10,108 +11,135 @@ RESOURCE_GROUP_NAME="$TEST_RESOURCE_PREFIX-$(date +%s)"
+  echo "VHD test VM username: $Test_VM_ADMIN_USERNAME, password: $TEST_VM_ADMIN_PASSWORD"
 
 else
-    az disk create --resource-group $RESOURCE_GROUP_NAME \
-        --name $DISK_NAME \
-        --source "${OS_DISK_URI}"  \
-        --query id
-    az vm create --name $VM_NAME \
-        --resource-group $RESOURCE_GROUP_NAME \
-        --attach-os-disk $DISK_NAME \
-        --os-type $OS_TYPE  \
-        --public-ip-address ""
+  az disk create --resource-group $RESOURCE_GROUP_NAME \
+    --name $DISK_NAME \
+    --source "${OS_DISK_URI}" \
+    --query id
+  az vm create --name $VM_NAME \
+    --resource-group $RESOURCE_GROUP_NAME \
+    --attach-os-disk $DISK_NAME \
+    --os-type $OS_TYPE \
+    --public-ip-address ""
 fi
 
 time az vm wait -g $RESOURCE_GROUP_NAME -n $VM_NAME --created
 
 FULL_PATH=$(realpath $0)
 CDIR=$(dirname $FULL_PATH)
-if [ "$OS_TYPE" == "Windows" ]; then
-    SCRIPT_PATH="$CDIR/$WIN_SCRIPT_PATH"
-    ret=$(az vm run-command invoke --command-id RunPowerShellScript \
-        --name $VM_NAME \
-        --resource-group $RESOURCE_GROUP_NAME  \
-        --scripts  @$SCRIPT_PATH \
-        --output json \
-        --parameters "containerRuntime=${CONTAINER_RUNTIME}" "WindowsSKU=${WINDOWS_SKU}")
-fi
-# An example of failed run-command output:
-# {
-#   "value": [
-#     {
-#       "code": "ComponentStatus/StdOut/succeeded",
-#       "displayStatus": "Provisioning succeeded",
-#       "level": "Info",
-#       "message": "c:\akse-cache\containerd\containerd-0.0.87-public.zip is cached as expected
-# c:\akse-cache\win-vnet-cni\azure-vnet-cni-singletenancy-windows-amd64-v1.1.2.zip is cached as expected
-# ... ...
-# "
-#       "time": null
-#     },
-#     {
-#       "code": "ComponentStatus/StdErr/succeeded",
-#       "displayStatus": "Provisioning succeeded",
-#       "level": "Info",
-#       "message": "Test-FilesToCacheOnVHD : File c:\akse-cache\win-k8s\v1.15.10-azs-1int.zip does not exist
-# At C:\Packages\Plugins\Microsoft.CPlat.Core.RunCommandWindows\1.1.5\Downloads\script0.ps1:146 char:1
-# + Test-FilesToCacheOnVHD
-# + ~~~~~~~~~~~~~~~~~~~~~~
-#     + CategoryInfo          : NotSpecified: (:) [Write-Error], WriteErrorException
-#     + FullyQualifiedErrorId : Microsoft.PowerShell.Commands.WriteErrorException,Test-FilesToCacheOnVHD
-#  ",
-#       "time": null
-#     }
-#   ]
-# }
 
-# we have to use `-E` to disable interpretation of backslash escape sequences, for jq cannot process string 
-# with a range of control characters not escaped as shown in the error below:
-#   Invalid string: control characters from U+0000 through U+001F must be escaped
-errMsg=$(echo -E $ret | jq '.value[]  | select(.code == "ComponentStatus/StdErr/succeeded") | .message')
-# a successful errMsg should be '""' after parsed by `jq`
-if [[ $errMsg != \"\" ]]; then
+if [ "$OS_TYPE" == "Linux" ]; then
+  SCRIPT_PATH="$CDIR/$LINUX_SCRIPT_PATH"
+  sleep 900
```

Comment:
az vm run-command does not work if we don't sleep here, I don't exactly know why it's taking so much time for the VM to be "run-command ready", but feel like this may be a series performance issue

Reply:
Do we need to wait more than 10 minutes after we get a ready VM? I did not find this issue for Windows case. In windows I hit a problem that docker is not ready when try to compare the images.

Reply:
as of now, this is the case, the run command keeps on retrying until it times out, so that's why I have added an extra 15minutes so it doesn't time out, but I'm investigating this issue why it is happening

Reply:
confirmed that this is not the case if we use a sig image then create VM for any region even for eastus.

Reply:
still trying to create the very same scenario for another region

Reply:
is it possible to check the VM status? i.e. polling the status until it is ready

Reply:
another optimization could be wrap the "az vm run-command" into a retry logic with timeout, check retrycmd_if_failure for example.

Reply:
tested this for couple of other regions as well, so looks like this is a common pattern. However, this does not happen for VMs created from SIG images.
@bowang-666 VM status is set to succeeded right away so querying the status won't help us. However, we may retry.

Reply:
Sounds good. Please change the sleep to retry, which will allow it to end dynamically.

Reply:
Done